src/ipv6/ipv6_wire.ml

```diff
@@ -103,6 +103,15 @@ let get_opt_prefix_on_link buf =
 let get_opt_prefix_autonomous buf =
   get_opt_prefix_reserved1 buf land 0x40 <> 0
 
+[%%cstruct
+type opt_rdnss = {
+    ty:                 uint8_t;
+    len:                uint8_t;
```

Comment:
hereafter we need a `reserved : uint16_t` (since there's some padding, and the lifetime starts at offset 4)